crates/emitter/src/emitter.rs

```diff
@@ -124,6 +124,17 @@ impl From<BytecodeOffset> for usize {
     }
 }
 
+pub struct BytecodeOffsetDiff {
+    pub diff: i32,
+}
+
+impl BytecodeOffsetDiff {
+    pub fn new(offset1: BytecodeOffset, offset2: BytecodeOffset) -> Self {
```

Comment:
it's not clear which is `from` and which is `to`.
how about adding a method to `BytecodeOffset` instead, like `let diff = offset1.diff_from(offset2);` and make the constructor of `BytecodeOffsetDiff` private?

also, it might be nice to move this into another module, so that private things are not accessible from other functions in this file.

then, for placeholder use case, I think you can add `BytecodeOffsetDiff::empty` or `BytecodeOffsetDiff::uninitialized` or something that fits the callsite